### ISSUE TITLE
Existing VPC template fails when using private instances and NAT gateways

### ISSUE THREAD

**Issue by philchristensen** [aws-quickstart/quickstart-redhat-openshift#15]
I'm having the same issue described in #6 despite having used the latest template from this repo. Things work pretty well until I get to this point:

```
Sep 17 02:25:29 localhost cloud-init: TASK [openshift_master : Start and enable master api all masters] **************
Sep 17 02:25:29 localhost cloud-init: skipping: [10.252.14.87]
Sep 17 02:25:30 localhost cloud-init: changed: [10.252.11.206]
Sep 17 02:25:30 localhost cloud-init: changed: [10.252.8.153]
Sep 17 02:25:31 localhost cloud-init: TASK [openshift_master : Dump logs from master-api if it failed] ***************
Sep 17 02:25:31 localhost cloud-init: skipping: [10.252.14.87]
Sep 17 02:25:31 localhost cloud-init: skipping: [10.252.8.153]
Sep 17 02:25:31 localhost cloud-init: skipping: [10.252.11.206]
Sep 17 02:25:31 localhost cloud-init: TASK [openshift_master : set_fact] *********************************************
Sep 17 02:25:31 localhost cloud-init: skipping: [10.252.14.87]
Sep 17 02:25:31 localhost cloud-init: ok: [10.252.8.153]
Sep 17 02:25:31 localhost cloud-init: ok: [10.252.11.206]
Sep 17 02:25:31 localhost cloud-init: TASK [openshift_master : Wait for API to become available] *********************
Sep 17 02:29:48 localhost cloud-init: FAILED - RETRYING: Wait for API to become available (120 retries left).
Sep 17 02:33:23 localhost dhclient[604]: DHCPREQUEST on eth0 to 10.252.10.1 port 67 (xid=0x63f783f2)
Sep 17 02:33:23 localhost dhclient[604]: DHCPACK from 10.252.10.1 (xid=0x63f783f2)
Sep 17 02:33:23 localhost NetworkManager[566]: <info>  [1505615603.1036] dhcp4 (eth0):   address 10.252.10.164
Sep 17 02:33:23 localhost NetworkManager[566]: <info>  [1505615603.1038] dhcp4 (eth0):   plen 24 (255.255.255.0)
Sep 17 02:33:23 localhost NetworkManager[566]: <info>  [1505615603.1038] dhcp4 (eth0):   gateway 10.252.10.1
Sep 17 02:33:23 localhost NetworkManager[566]: <info>  [1505615603.1038] dhcp4 (eth0):   lease time 3600
Sep 17 02:33:23 localhost NetworkManager[566]: <info>  [1505615603.1038] dhcp4 (eth0):   hostname 'ip-10-252-10-164'
Sep 17 02:33:23 localhost NetworkManager[566]: <info>  [1505615603.1038] dhcp4 (eth0):   nameserver '10.252.8.2'
Sep 17 02:33:23 localhost NetworkManager[566]: <info>  [1505615603.1038] dhcp4 (eth0):   domain name 'aws.logicworks.local'
Sep 17 02:33:23 localhost NetworkManager[566]: <info>  [1505615603.1039] dhcp4 (eth0): state changed bound -> bound
Sep 17 02:33:23 localhost dbus[555]: [system] Activating via systemd: service name='org.freedesktop.nm_dispatcher' unit='dbus-org.freedesktop.nm-dispatcher.service'
Sep 17 02:33:23 localhost dbus-daemon: dbus[555]: [system] Activating via systemd: service name='org.freedesktop.nm_dispatcher' unit='dbus-org.freedesktop.nm-dispatcher.service'
Sep 17 02:33:23 localhost dhclient[604]: bound to 10.252.10.164 -- renewal in 1515 seconds.
Sep 17 02:33:23 localhost systemd: Starting Network Manager Script Dispatcher Service...
Sep 17 02:33:23 localhost dbus-daemon: dbus[555]: [system] Successfully activated service 'org.freedesktop.nm_dispatcher'
Sep 17 02:33:23 localhost dbus[555]: [system] Successfully activated service 'org.freedesktop.nm_dispatcher'
Sep 17 02:33:23 localhost nm-dispatcher: req:1 'dhcp4-change' [eth0]: new request (4 scripts)
Sep 17 02:33:23 localhost nm-dispatcher: req:1 'dhcp4-change' [eth0]: start running ordered scripts...
Sep 17 02:33:23 localhost systemd: Started Network Manager Script Dispatcher Service.
Sep 17 02:34:06 localhost cloud-init: FAILED - RETRYING: Wait for API to become available (119 retries left).
Sep 17 02:38:24 localhost cloud-init: FAILED - RETRYING: Wait for API to become available (118 retries left).
Sep 17 02:42:42 localhost cloud-init: FAILED - RETRYING: Wait for API to become available (117 retries left).
```

I've waited more than 30 retries, but no dice. Any help would be appreciated.

**Comment by philchristensen** [comment]
If I had to guess, I'd say it was kind of certificate error, since if I go to the health check URL on the master and ignore the cert warnings, I get 'ok' as expected.

**Comment by philchristensen** [comment]
It could also be a DNS issue. It would be great to know what the installer/distribution expects the DNS resolution situation to be.

Some k8s distributions have had issues that the automatically-defined AWS hostnames are not currently resolvable in my local DNS. On Tectonic I add `--kubelet-preferred-address-types=InternalIP` to make things work properly.

**Comment by philchristensen** [comment]
Ah, I've found the issue. The OpenShift security group didn't contain a rule for the outgoing IP of the NAT gateway, or the public IP of the ansible node, so when used as an ELB group, it wasn't allowing access to the API.

**Comment by avattathil** [comment]
@philchristensen thanks for reporting this. I just wanted to confirm that you were using the Subscription-Version of the playbook.

**Comment by philchristensen** [comment]
Yes, I used the defaults in the template, which were Subscription-Version and repo tag 3.6.173.0.5-5

**Comment by avattathil** [comment]
@philchristensen  I was able to recreate the issue you are seeing. I could not replicate this issues initially because I was not limiting the `AccessCIDR`.  Your suggestion should solve the problem.

**Comment by avattathil** [comment]
@philchristensen I updated the master branch with a fix to the issue you reported. After doing some research I realized the timeouts were occurring due to overrides to the cluster hostname which routed the API call incorrectly 😠  (I have an initial fix in place)

Commit https://github.com/aws-quickstart/quickstart-redhat-openshift/commit/4cf2ee298953ce980d372dce672ec911d67f060d Instructs the cluster to use the internal address which resolves the issue. I am working on a few updates that should make the template more robust. Let me know that this initial fix works for you.
